functions/venue.js

```diff
@@ -711,31 +711,26 @@ exports.deleteTable = functions.https.onCall(async (data, context) => {
   const tables = docData.config.tables || defaultTables;
 
   const index = tables.findIndex((val) => val.reference === tableName);
-  // there are venues that don't have tables inserted in the database and use default ones instead
-  // if that's the case, then we won't be able to delete any of the tables
-  // thus we have to create tables (excluding the one being deleted) before modifying them
-  if (!docData.config || !docData.config.tables) {
-    defaultTables.splice(index, 1);
-
-    admin
-      .firestore()
-      .collection("venues")
-      .doc(spaceId)
-      .update({
-        ...docData,
-        config: { ...docData.config, tables: defaultTables },
-      });
-
-    return;
-  }
 
   if (index === -1) {
     throw new HttpsError("not-found", `Table does not exist`);
-  } else {
-    docData.config.tables.splice(index, 1);
   }
 
-  admin.firestore().collection("venues").doc(spaceId).update(docData);
+  tables.splice(index, 1);
+
+  const recalculatedTables = tables.map((table, i) => ({
+    ...table,
+    reference: `Table ${i + 1}`,
```

Comment:
I think remapping the references might be a bad idea, and it can lead to edge cases. I think everything is fine except for this part, maybe we should use uuid for new table references and keep the old ones as they are.

What happens if you and me join a table and at that point in time the reference is `Table 10` after that someone removes another table, adds a new one and our table reference becomes `Table 9` If a new person tries to join our table with you and me, he will probably join another reference and be alone, it's a wild guess, but sounds like this is going to happen.

Reply:
fixed 131ac0b